src/gsi/guess_grids.F90

```diff
@@ -1745,13 +1745,19 @@ subroutine load_gsdpbl_hgt(mype)
 
              do k=1,nsig
 
-                if (wrf_mass_regional)  pbk(k) = aeta1_ll(k)*(ges_ps_01(i,j)*ten-pt_ll)+aeta2_ll(k)+pt_ll
-		if (nems_nmmb_regional) then
-		   pbk(k) = aeta1_ll(k)*pdtop_ll + aeta2_ll(k)*(ten*ges_ps(i,j) & 
-		            -pdtop_ll-pt_ll) + pt_ll   			    			    
-		end if
-				
-		thetav(k)  = ges_tv(i,j,k)*(r1000/pbk(k))**rd_over_cp_mass
+                if (wrf_mass_regional) then
```

Comment:
This part bring significant changes .  I suspect this part (for fv3_regional) also caused the differences in my comparison test, in which , the new codes with this subdomain function is supposed to give the identical results for the single domain case.  Would @hu5970 give some documentation  on why the changes are expected to be more accurate?